tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -2535,3 +2541,68 @@ void SwiftLangSupport::editorExpandPlaceholder(StringRef Name, unsigned Offset,
   }
   EditorDoc->expandPlaceholder(Offset, Length, Consumer);
 }
+
+//===----------------------------------------------------------------------===//
+// Semantic Tokens
+//===----------------------------------------------------------------------===//
+
+void SwiftLangSupport::getSemanticTokens(
+    StringRef PrimaryFilePath, StringRef InputBufferName,
+    ArrayRef<const char *> Args, llvm::Optional<VFSOptions> VfsOptions,
+    SourceKitCancellationToken CancellationToken,
+    std::function<void(const RequestResult<SemanticTokensResult> &)> Receiver) {
+  std::string FileSystemError;
+  auto FileSystem = getFileSystem(VfsOptions, PrimaryFilePath, FileSystemError);
+  if (!FileSystem) {
+    Receiver(RequestResult<SemanticTokensResult>::fromError(FileSystemError));
+    return;
+  }
+
+  std::string InvocationError;
+  SwiftInvocationRef Invok = ASTMgr->getTypecheckInvocation(
+      Args, PrimaryFilePath, FileSystem, InvocationError);
+  if (!InvocationError.empty()) {
+    LOG_WARN_FUNC("error creating ASTInvocation: " << InvocationError);
+  }
```

Comment:
Where does this actually show up? Is it ever more obvious than just the actual error response?

Reply:
This is copied from 

https://github.com/apple/swift/blob/4ac34a40ea9e3193be342b58ad2e1abb2399a155/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp#L2110

and 

https://github.com/apple/swift/blob/4ac34a40ea9e3193be342b58ad2e1abb2399a155/tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp#L2216C20-L2216C27

Not sure if it’s more obvious but it’s also not less obvious so 🤷🏽